src/Common/Site_Health/Info_Field_Abstract.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace TEC\Common\Site_Health;
+
+/**
+ * Class Info_Field_Abstract
+ *
+ * @link    https://developer.wordpress.org/reference/hooks/debug_information/
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Site_Health
+ */
+abstract class Info_Field_Abstract implements Info_Field_Interface {
+	/**
+	 * Stores the ID for the field.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected string $id;
+
+	/**
+	 * Stores the label for the field.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected string $label;
+
+	/**
+	 * Stores the value for field.
+	 *
+	 * @since TBD
+	 *
+	 * @return string|int|float|array<int>|array<float>|array<string>
+	 */
+	protected $value;
+
+	/**
+	 * Stores the priority for the field, used for sorting.
+	 *
+	 * @since TBD
+	 *
+	 * @var int
+	 */
+	protected int $priority;
+
+	/**
+	 * Stores the flag for if the field is private.
+	 *
+	 * @since TBD
+	 *
+	 * @var bool
+	 */
+	protected bool $is_private;
+
+	/**
+	 * Stores the debug value for the field.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected string $debug;
+
+	/**
+	 * @inheritDoc
+	 */
+	public function get_id(): string {
+		return $this->id;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function get_label(): string {
+		return $this->label;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function get_value() {
+		return $this->value;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function get_priority(): int {
+		return $this->priority;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function is_private(): bool {
+		return $this->is_private;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function get_debug(): string {
+		return $this->debug;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function to_array( Info_Section_Abstract $section ): array {
```

Comment:
```suggestion
	public function to_array( Info_Section_Abstract $section ): array {
```
Not as familiar with all of the structure for this, but this approach is a bit odd to me. Pass a `$section` object in order to get an "internal" fields `array` structure? Seems like it should be more strictly defined.

Perhaps it's as simple as renaming this function to help, or these filters run elsewhere? But it begs the question if the relationship between section/field needs to be associated/deassociated more? Maybe with `Field::section` or `$section->get_fields()`... 

Hope the point I'm trying to make is clear?